runtime/codert_vm/cnathelp.cpp

```diff
@@ -466,6 +466,37 @@ fast_jitNewObjectImpl(J9VMThread *currentThread, J9Class *objectClass, bool chec
 	goto done;
 }
 
+static VMINLINE bool
+fast_jitNewValueImpl(J9VMThread *currentThread, J9Class *objectClass, bool checkClassInit, bool nonZeroTLH)
+{
+	bool slowPathRequired = false;
+	if (checkClassInit) {
+		if (J9_UNEXPECTED(VM_VMHelpers::classRequiresInitialization(currentThread, objectClass))) {
+			goto slow;
+		}
+	}
+	if (J9_UNEXPECTED(!J9_IS_J9CLASS_VALUETYPE(objectClass))) {
+		goto slow;
+	}
+	{
+		UDATA allocationFlags = J9_GC_ALLOCATE_OBJECT_INSTRUMENTABLE;
+		if (nonZeroTLH) {
+			allocationFlags |= J9_GC_ALLOCATE_OBJECT_NON_ZERO_TLH;
+		}
+		j9object_t obj = currentThread->javaVM->memoryManagerFunctions->J9AllocateObjectNoGC(currentThread, objectClass, allocationFlags);
+		if (J9_UNEXPECTED(NULL == obj)) {
+			goto slow;
+		}
+		JIT_RETURN_UDATA(obj);
+	}
+done:
+	return slowPathRequired;
+slow:
+	currentThread->floatTemp1 = (void*)objectClass;
```

Comment:
Doesn't have to be fixed here but noting to others that this `floatTemp1` field has become quite overloaded.

Reply:
No more than they already were (this is a copy of the existing new object code). Individual fields would be nice, but we need to be careful not to push important vmthread fields out of the immediate constant range (or small constant range on X) otherwise the hand-written assembly becomes a lot nastier.